tests/unit/test-cmd/test.run.js

```diff
@@ -325,5 +326,47 @@ describe('run', () => {
            sourceDir: '/fake/source/dir',
          },
        );
-     });
-});
+     }
+  );
+
+  it('creates dir when profile doesn\'t exist using profile-create-new',
+     async () => {
+       sinon.stub(fs, 'existsSync').returns(false);
+       sinon.stub(fs, 'mkdirSync');
+
+       const firefoxProfile = '/pretend/path/to/Firefox/profile';
+       const cmd = prepareRun();
+
+       await cmd.run({
+         firefoxProfile,
+         profileCreateNew: true,
+       });
+
+       sinon.assert.calledWith(fs.mkdirSync, firefoxProfile);
+
+       fs.existsSync.restore();
+       fs.mkdirSync.restore();
+     }
+  );
+
+  it('throws error when profile dir already exists using profile-create-new',
+     async () => {
+       sinon.stub(fs, 'existsSync').returns(true);
+
+       const firefoxProfile = '/pretend/path/to/Firefox/profile';
+       const cmd = prepareRun();
+
+       const promise = cmd.run({
+         firefoxProfile,
+         profileCreateNew: true,
+       });
+
+       await assert.isRejected(
+         promise,
+         `Directory ${firefoxProfile} already exists.`
+       );
+
+       fs.existsSync.restore();
+     }
+  );
```

Comment:
Should I also add a test that covers the use case of `--chromium-profile` with `--profile-create-new`?

Reply:
@akhilpanchal for the unit test I think that it does make sense to also assert the expected behavior when passing `chromiumProfile` to cmd.run instead of `firefoxProfile`.

I'm actually curious about: what is the current  `web-ext run -t chromium --chromium-profile ...` behavior (without any of these changes) when the given profile directory doesn't exist?

Reply:
@rpl 
Sure. I can add more tests to cover the case for `--chromium-profile` with `--profile-create-new`.

Following is the output for the `web-ext run -t chromium --chromium-profile ...` command without any of these changes:

```sh
$ web-ext run -t chromium --chromium-profile /Users/apanchal/MyCode/extension-borderify/foobar
Running web extension from /Users/apanchal/MyCode/extension-borderify

Error: ENOENT: no such file or directory, stat '/Users/apanchal/MyCode/extension-borderify/foobar'

Error code: ENOENT
```